### PR TITLE
Low memory optimisations

### PR DESCRIPTION
**Pull request by miohtama**
- Offer an option to load Parquet files with filtering (not used)
- Load candles data over new JSONL endpoint

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,6 +55,7 @@ sphinx = "^4.2.0"
 sphinx-sitemap = "^2.2.0"
 coloredlogs = "^15.0.1"
 
+memory-profiler = "^0.60.0"
```
Comment (hieuh25) [inline pyproject.toml:58]: We probably don't need this as dependency

File: tests/test_candle_universe.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
+import resource
+
+import pandas
```
Comment (hieuh25) [inline tests/test_candle_universe.py:3]: Duplicated import

File: tradingstrategy/client.py [hunk 3]
```diff
@@ -7,17 +7,19 @@
...
 from functools import wraps
 from json import JSONDecodeError
-from typing import Optional
+from typing import Optional, Set
 from pathlib import Path
 import logging
 
 # TODO: Must be here because  warnings are very inconveniently triggered import time
+import pandas as pd
+from backtrader import List
```
Comment (hieuh25) [inline tradingstrategy/client.py:22]: I have a feeling this is the wrong (automatic) import, should it be from `typing`?

File: tradingstrategy/transport/cache.py [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+import hashlib
 import io
 import os
 import pathlib
-from typing import Optional, Callable
+from typing import Optional, Callable, List, Set
 import shutil
 import logging
 
+import pandas
```
Comment (hieuh25) [inline tradingstrategy/transport/cache.py:10]: Duplicated import

**Review by hieuh25 — APPROVED** [review]
Looking good!

I made a few comments there and there

File: tradingstrategy/client.py [hunk 5]
```diff
@@ -164,6 +166,58 @@ def fetch_all_candles(self, bucket: TimeBucket) -> pyarrow.Table:
...
+            start_time,
+            end_time,
+            max_bytes=max_bytes,
+        )
+
+    def fetch_candle_dataset(self, bucket: TimeBucket) -> Path:
+        """Fetch candle data from the server.
+
+        Do not attempt to decode the Parquet file to the memory,
+        but instead of return raw
```
Comment (plamut) [inline tradingstrategy/client.py:216]: Missing end of the sentence? :)

**Review by plamut — COMMENTED** [review]
Filtering should help immensely, which is great!

Spotted a few cleanup opportunities, but nothing major really.

**Pull request merged**